a10_octavia/controller/worker/tasks/virtual_port_tasks.py

```diff
@@ -244,3 +247,35 @@ def execute(self, loadbalancer, listener, vthunder):
         except (acos_errors.ACOSException, ConnectionError) as e:
             LOG.exception("Failed to delete listener: %s", listener.id)
             raise e
+
+
+class GetListenersStats(ListenersParent, task.Task):
+    """Task for retrieving listener stats from vthunder"""
+
+    @axapi_client_statistics_decorator
+    def execute(self, vthunder, listeners):
+        listener_stats = []
+        if listeners:
+            for listener in listeners:
+                listener.protocol = openstack_mappings.virtual_port_protocol(self.axapi_client,
```

Comment:
https://github.com/a10networks/acos-client/blob/master/acos_client/v30/slb/virtual_server.py#L104

This iterative call on the API shouldn't be needed. Should be able to get all the port stats at once. If it can't and returns an error then it needs to be removed as part of https://github.com/a10networks/acos-client/pull/355/files

Reply:
As discussed over teams,

Now calling axapi for retrieving the stats of load balancer, then iterating over the response of axapi to get the stats for each listener.